src/libdredd/src/mutation_replace_unary_operator.cc

```diff
@@ -0,0 +1,263 @@
+                   << "arg();\n";
+    } else {
+      new_function << "arg()" << clang::UnaryOperator::getOpcodeStr(op).str()
+                   << ";\n";
+    }
+    mutant_offset++;
+  }
+  new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+               << mutant_offset << ")) return arg();\n";
+  mutant_offset++;
+
+  if (unary_operator_.getOpcode() == clang::UnaryOperatorKind::UO_LNot) {
+    // true
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return true;\n";
+    mutant_offset++;
+
+    // false
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return false;\n";
+    mutant_offset++;
+  }
+
+  new_function << "  return ";
+  if (IsPrefix(unary_operator_.getOpcode())) {
+    new_function
+        << clang::UnaryOperator::getOpcodeStr(unary_operator_.getOpcode()).str()
+        << "arg();\n";
+  } else {
+    new_function
+        << "arg()"
+        << clang::UnaryOperator::getOpcodeStr(unary_operator_.getOpcode()).str()
+        << ";\n";
+  }
+  new_function << "}\n\n";
+
+  // The function captures |mutant_offset| different mutations, so bump up
+  // the mutation id accordingly.
+  mutation_id += mutant_offset;
+
+  return new_function.str();
+}
+
+void MutationReplaceUnaryOperator::ApplyTypeModifiers(const clang::Expr* expr,
+                                                      std::string& type) {
+  if (expr->isLValue()) {
+    type += "&";
+    clang::QualType qualified_type = expr->getType();
+    if (qualified_type.isVolatileQualified()) {
+      assert(expr->getType().isVolatileQualified() &&
+             "Expected expression to be volatile-qualified since subexpression "
+             "is.");
+      type = "volatile " + type;
+    }
+  }
+}
+
+void MutationReplaceUnaryOperator::Apply(
+    clang::ASTContext& ast_context, const clang::Preprocessor& preprocessor,
+    int first_mutation_id_in_file, int& mutation_id, clang::Rewriter& rewriter,
+    std::unordered_set<std::string>& dredd_declarations) const {
+  std::string new_function_name = GetFunctionName(ast_context);
+  std::string result_type = unary_operator_.getType()
+                                ->getAs<clang::BuiltinType>()
+                                ->getName(ast_context.getPrintingPolicy())
+                                .str();
+  std::string input_type = unary_operator_.getSubExpr()
+                               ->getType()
+                               ->getAs<clang::BuiltinType>()
+                               ->getName(ast_context.getPrintingPolicy())
+                               .str();
+
+  // To avoid problems of ambiguous function calls, the argument types (ignoring
+  // whether they are references or not) are baked into the mutation function
+  // name. Some type names have space in them (e.g. 'unsigned int'); such spaces
+  // are replaced with underscores.
```

Comment:
This comment applies to the code that you've extracted out into GetFunctionName, so it should be moved there.

If this applies to the binary operator case also then it would be fine to adapt that file in this PR.

Reply:
Moved.